File: ngSkillVilla/src/app/models/post.ts
import { Community } from './community';
import { Location } from './location';
import { PostCategory} from './post-category';
import { User } from './user';
export class Post {
    id: number;
    user: User;
    community: Community | null;
    description: string;
    enabled: boolean;
    location: Location | null;
    postCategory: PostCategory;
    createdAt: string;
    updatedAt: string;
    
    constructor(
        id = 0,
        user = new User(),
        community = null,
        description = '',
        enabled = false,
        location = null,
        postCategory = new PostCategory(),
        createdAt = '',
        UpdatedAt = ''
    ) {
        this.id = id;
        this.user = user;
        this.community = community;
        this.description = description;
        this.enabled = enabled;
        this.location = location;
        this.postCategory = postCategory;
        this.createdAt = createdAt;
        this.updatedAt = UpdatedAt;
    }

}

File: SkillVillaJPA/src/main/java/com/skilldistillery/skillvilla/entities/Post.java
package com.skilldistillery.skillvilla.entities;

import java.time.LocalDateTime;
import java.util.Objects;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

@Entity
public class Post {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	@Column(name="created_at")
	private LocalDateTime createdAt;

	@Column(name="updated_at")
	private LocalDateTime updatedAt;
	
	private String description;
	
	private boolean enabled;
	
//  private int userId;
	
//  private int postId;

//  private int locationId;
	
	public Post () {	
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public LocalDateTime getUpdatedAt() {
		return updatedAt;
	}

	public void setUpdatedAt(LocalDateTime updatedAt) {
		this.updatedAt = updatedAt;
	}

	public LocalDateTime getCreatedAt() {
		return createdAt;
	}

	public void setCreatedAt(LocalDateTime createdAt) {
		this.createdAt = createdAt;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public boolean isEnabled() {
		return enabled;
	}

	public void setEnabled(boolean enabled) {
		this.enabled = enabled;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Post other = (Post) obj;
		return id == other.id;
	}

	@Override
	public String toString() {
		return "Post [id=" + id + ", updatedAt=" + updatedAt + ", createdAt=" + createdAt + ", description="
				+ description + ", enabled=" + enabled + "]";
	};

	
	
}

